FinSearchUnified/tests/Helper/StaticHelperTest.php

```diff
@@ -152,4 +190,29 @@ function testUseShopSearch(
         $shop = $expectedResult ? "shop" : "FINDOLOGIC";
         $this->assertEquals($result, $expectedResult, sprintf($error, $shop));
     }
+
+    /**
+     * @dataProvider controlCharacterProvider
+     *
+     * @param string $text
+     * @param string $expected
+     */
+    public function testControlCharacterMethod($text, $expected)
+    {
+        $result = StaticHelper::removeControlCharacters($text);
+        $this->assertEquals($expected, $result, sprintf('%s was expected but %s was returned', $expected, $result));
+    }
+
+    /**
+     * @dataProvider cleanStringProvider
+     * @dataProvider controlCharacterProvider
```

Comment:
Interesting. Are these data providers executed sequentially?

Reply:
Yes but they are merged together and then executed together as if they are part of a single provider